### PR TITLE
feat: The Graph migration (moving on from GoldSky)

### PR DESCRIPTION
**Pull request by wsdt**
All subgraphs deployed. If approved then will release new major version and propose that we extensively test the gateway on staging and deploy lightbridge for testnet with the update. 

Make sure to set `THE_GRAPH_API_KEY` in your .env when using this package in your app.

### CODE REVIEW

**Comment by boyuan-chen** [comment]
@wsdt Could you please solve the conflicts?

File: src/graphql.service.ts [hunk 1]
```diff
@@ -5,115 +5,128 @@ import {
...
         // BSC
         56: {
             [EGraphQLService.LightBridge]: {
-                gql: 'https://api.goldsky.com/api/public/project_clq6jph4q9t2p01uja7p1f0c3/subgraphs/light-bridge-bsc/v1/gn',
+                gql: this.withSubgraphId('4Eoi62aX8V1DCebNX4mzPi7rfceFaRAeZ3xt7AYVowpy'),
                 local: '',
             },
         },
         // BNB testnet
         97: {
```
Comment (sk-enya) [inline src/graphql.service.ts:85]: can we have explorer for BNB testnet `AnchorageBridge` so we can query the proven transactions and finalized

**Review by sk-enya — APPROVED** [review]
Looks good!

**Comment by wsdt** [comment]
will test all this today with the Gateway / LightBridge first.

File: src/graphql.service.ts [hunk 2]
```diff
@@ -5,115 +5,136 @@ import {
...
+config();
 
 let fetchLib = fetch
 if (!fetchLib) {
     fetchLib = require('node-fetch');
 }
 
-
 export class GraphQLService {
+    private readonly apikey = process.env.THE_GRAPH_API_KEY ?? '815ddde7e44ffb91132567e2c4e7c558'
```
Comment (boyuan-chen) [inline src/graphql.service.ts]: Is this the API key?

File: src/graphql.service.ts [hunk 3]
```diff
@@ -5,136 +5,157 @@ import {
...
         // Boba BNB testnet
         9728: {
             [EGraphQLService.LightBridge]: {
-                gql: 'https://api.goldsky.com/api/public/project_clq6jph4q9t2p01uja7p1f0c3/subgraphs/light-bridge-boba-bnb-testnet/v1/gn',
+                gql: 'https://graphql.testnet.bnb.boba.network/subgraphs/name/boba-bnb-l2-testnet/',
                 local: 'http://127.0.0.1:8002/subgraphs/name/boba/Bridges',
             },
+            [EGraphQLService.AnchorageBridge]: {
+                gql: 'https://graphql.bnb.boba.network/subgraphs/name/anchorage-bridging-boba-bnb',
+            },
```
Comment (sk-enya) [inline src/graphql.service.ts]: @wsdt looks like subgraphs is refered as mainnet one like Boba BNB above.

https://github.com/bobanetwork/graphql-utils/pull/6/files#diff-598c468d2a6aebfb01813d5057757685113469de2752fb80cadc288859448389R73-R75

**Review by sk-enya — CHANGES_REQUESTED** [review]

File: src/graphql.service.ts (hunk 3 above)
Comment (wsdt) [reply src/graphql.service.ts]: @sk-enya what do you mean with is referred as mainnet? It is within the boba bnb mainnet config

Comment (sk-enya) [reply src/graphql.service.ts]: I mean this url is for BOBA bnb mainnet subgraph.!

**Review by sk-enya — APPROVED** [review]

**Pull request merged**